File: frontend/src/types/ThesisSchema.ts
export interface ThesisSchema {
  cover: {
    titleZh: string;
    titleEn: string;
    department: string;
    degree: '博士' | '碩士';
    studentName: string;
    advisorName: string;
    year: string;
    month: string;
  };

  abstractZh: {
    content: string;
    keywords: string[];
  };

  abstractEn: {
    content: string;
    keywords: string[];
  };

  acknowledgments: string;
  chapters: Chapter[];
  bibliography: string[];
  appendices?: Appendix[];
  figures: FigureEntry[];
  tables: TableEntry[];
  symbols?: SymbolEntry[];
}

export interface Chapter {
  number: number;
  titleZh: string;
  titleEn: string;
  sections: Section[];
}

export interface Section {
  id: string;
  titleZh: string;
  titleEn?: string;
  content: string;
  subsections?: Section[];
}

export interface FigureEntry {
  number: number;
  title: string;
  page?: number;
}

export interface TableEntry {
  number: number;
  title: string;
  page?: number;
}

export interface SymbolEntry {
  symbol: string;
  description: string;
}

export interface Appendix {
  label: string;
  title: string;
  content: string;
}

File: backend/schemas/thesis_schema.py
from __future__ import annotations
from typing import Literal, Optional
from pydantic import BaseModel, Field


class Section(BaseModel):
    id: str
    titleZh: str
    titleEn: Optional[str] = ""
    content: str
    subsections: Optional[list[Section]] = None


class Chapter(BaseModel):
    number: int
    titleZh: str
    titleEn: str
    sections: list[Section]


class FigureEntry(BaseModel):
    number: int
    title: str
    page: Optional[int] = None


class TableEntry(BaseModel):
    number: int
    title: str
    page: Optional[int] = None


class SymbolEntry(BaseModel):
    symbol: str
    description: str


class Appendix(BaseModel):
    label: str
    title: str
    content: str


class AbstractBlock(BaseModel):
    content: str
    keywords: list[str] = Field(default_factory=list)


class Cover(BaseModel):
    titleZh: str = ""
    titleEn: str = ""
    department: str = ""
    degree: Literal["博士", "碩士"] = "碩士"
    studentName: str = ""
    advisorName: str = ""
    year: str = ""
    month: str = ""


class ThesisSchema(BaseModel):
    cover: Cover = Field(default_factory=Cover)
    abstractZh: AbstractBlock = Field(default_factory=AbstractBlock)
    abstractEn: AbstractBlock = Field(default_factory=AbstractBlock)
    acknowledgments: str = ""
    chapters: list[Chapter] = Field(default_factory=list)
    bibliography: list[str] = Field(default_factory=list)
    appendices: Optional[list[Appendix]] = None
    figures: list[FigureEntry] = Field(default_factory=list)
    tables: list[TableEntry] = Field(default_factory=list)
    symbols: Optional[list[SymbolEntry]] = None
